ocean/src/clans.rs

```diff
@@ -1,39 +1,76 @@
+use std::collections::HashMap;
 
 #[derive(Debug)]
 pub struct ClanSystem {
     // TODO: add necessary fields
+    my_map: HashMap<String, Vec<String>>,
 }
 
 impl ClanSystem {
     pub fn new() -> ClanSystem {
-        unimplemented!();
+        ClanSystem{ my_map : HashMap::new() }
+    }
+
+    pub fn add_member_to_clan(&mut self, clan_id: &str, crab_name: &str) {
+        if(self.my_map.contains_key(clan_id)) {
+            let list = self.my_map.get_mut(clan_id);
+            list.unwrap().push(crab_name.to_string());
+        } else {
+            let mut list: Vec<String> = Vec::new();
+            list.push(crab_name.to_string());
+            self.my_map.insert(clan_id.to_string(), list);
+        }
+    }
+
+    pub fn contains_clan(&self, clan_id: &str) -> bool {
+        return self.my_map.contains_key(clan_id);
```

Comment:
I personally also like helper functions to simplify the code! But in general, though, one-line functions don't necessarily need to be abstracted out to a separate function.
Also side note, this contains_clan could be used in add_member_to_clan as well though, since I believe Rust method ordering doesn't matter in this scope.

Reply:
Also, this logic of contains_key is much better than the logic I used in my code - wish I knew about that method!